proof.go

```diff
@@ -141,6 +141,58 @@ func (proof Proof) IsEmptyProof() bool {
 	return proof.start == proof.end && len(proof.nodes) == 0 && len(proof.leafHash) == 0
 }
 
+// IsValidEmptyRangeProof checks whether the proof is a valid empty range proof.
+func (proof Proof) IsValidEmptyRangeProof(nth *NmtHasher, nID namespace.ID, root []byte, leaves [][]byte, checkNS bool) bool {
```

Comment:
why do we need this method and how empty proof can be valid or not? I see you are using it under if `proof.start == proof.end ` condition, but proof.start can't be equal to proof.end conceptually, because `end` is a non-inclusive index of the leave.

Reply:
I don't think we should check the validity of an empty proof at all.

Reply:
I think it was in previous implementation so ratianal is backwards compatability to previous behavior. This is meant as non-breaking PR, but we might want to consider to break it in following PR. 

@gupadhyaya  Perhaps we should make this func as unexported for now?

Reply:
will have to come back to this. for now, made this function private and retained the logic that was there previously. don't want to block the getrange PR on this. @vgonkivs

Reply:
have created an issue for coming back to this: https://github.com/celestiaorg/nmt/issues/292